shims/spark34/src/main/scala/org/apache/spark/sql/execution/FileSourceScanExecShim.scala

```diff
@@ -56,12 +56,15 @@ abstract class FileSourceScanExecShim(
     case FileSourceGeneratedMetadataAttribute(attr) => attr
   }
 
-  def dataFiltersInScan: Seq[Expression] = dataFilters
+  def dataFiltersInScan: Seq[Expression] = dataFilters.filterNot(_.references.exists {
```

Comment:
So we don't support filters against `row_index` column in Spark3.4? will it also apply to Spark3.5?

Reply:
it not means `don't support filters` against `row_index` column, it just move the `filter` from `TableScan` node to `filterProject` node.